compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -60,6 +61,13 @@ pub(crate) fn optimize_into_acir(
         .finish();
 
     let brillig = ssa.to_brillig(print_brillig_trace);
+
+    // Split off any passes the are not necessary for Brillig generation but are necessary for ACIR generation.
+    // We only need to fill out nested slices as we need to have a known length when dealing with memory operations
+    // in ACIR gen while this is not necessary in the Brillig IR.
+    let ssa = SsaBuilder::new_with_generated_ssa(ssa, print_ssa_passes)
+        .run_pass(Ssa::fill_internal_slices, "After Fill Internal Slice Dummy Data:")
+        .finish();
```

Comment:
This seems awkward to me. Instead of destroying the old SsaBuilder in the previous .finish call then recreating it here, how about we add a `to_brillig` method on the builder itself? Then we could do:

```rs
let ssa = SsaBuilder::new(...)
    .run_pass(.. pass 1 ..)
    ...
    .run_pass(.. pass N ..); // no .finish, we're not done

let brillig = ssa.to_brillig(print_brillig_trace);

let ssa = ssa.run_pass(Ssa::fill_internal_slices, ..).finish(); // now we're done
```